cli/popper/commands/cmd_env-vars.py

```diff
@@ -0,0 +1,84 @@
+import click
+import popper.utils as pu
+
+from popper.cli import pass_context
+from popper.exceptions import UsageError
+
+
+@click.command('matrix', short_help='Define execution matrix for a '
+               'pipeline using environment variables.')
+@click.argument('pipeline', required=True)
+@click.option(
+    '--add',
+    multiple=True,
+    help='Add environment variable in the form key=value',
+    required=False
+)
+@click.option(
+    '--rm',
+    multiple=True,
+    help='Remove environment variable using the key',
+    required=False
+)
+@pass_context
+def cli(ctx, pipeline, add, rm):
+    """Add or remove environment variables defined for matrix execution of
+    a pipeline."""
+
+    config = pu.read_pipeline_config(pipeline)
+
+    if add and rm:
+        raise UsageError("Both add and rm cannot be given at the same time. "
+                         "See popper env-vars --help for more information.")
+
+    if add:
+        try:  # This try catch block ensures backward compatibility
+            env_vars = config['vars']
+        except KeyError:
+            env_vars = []
+        vars_add = {}
+        for var in add:
+            key, val = var.split('=')
+            vars_add[key] = val
+        env_vars.append(vars_add)
+        pu.update_config(pipeline, vars=env_vars)
```

Comment:
If we make `pu.read_pipeline_config(pipeline)` return a tuple (see below), we can use `pu.write_config()` directly